src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java

```diff
@@ -0,0 +1,177 @@
+package mate.academy.internetshop.dao.jdbc;
+
+import mate.academy.internetshop.annotations.Dao;
+import mate.academy.internetshop.dao.BucketDao;
+import mate.academy.internetshop.model.Bucket;
+import mate.academy.internetshop.model.Item;
+import org.apache.log4j.Logger;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class BucketDaoJdbcImpl extends AbstractDao<Bucket> implements BucketDao {
+    private static final String TABLE_BUCKETS = "buckets";
+    private static final String TABLE_BUCKETS_ITEMS = "buckets_items";
+    private static final String TABLE_ITEMS = "items";
+    private static Logger logger = Logger.getLogger(BucketDaoJdbcImpl.class);
+
+    public BucketDaoJdbcImpl(Connection connection) {
+        super(connection);
+    }
+
+    @Override
+    public Bucket create(Bucket bucket) {
+        String queryToBuckets = String.format("insert into %s (user_id) values (?)", TABLE_BUCKETS);
+        try (PreparedStatement statement = connection.prepareStatement(
+                queryToBuckets, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, bucket.getOwnerID());
+            statement.executeUpdate();
+            ResultSet rs = statement.getGeneratedKeys();
+            while (rs.next()) {
+                bucket.setBucketId(rs.getLong(1));
+            }
+        } catch (SQLException e) {
+            logger.warn("Can't create bucket", e);
+        }
+        return insertToBucketsItems(bucket);
+    }
+
+    @Override
+    public Optional<Bucket> get(Long id) {
+        Bucket bucket = new Bucket(id);
+        String getUserIdQuery = String.format(
+                "select user_id from %s where bucket_id = ?;", TABLE_BUCKETS);
+        try (PreparedStatement statement = connection.prepareStatement(getUserIdQuery)) {
+            statement.setLong(1, id);
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
+                bucket.setOwnerId(rs.getLong(1));
+            }
+        } catch (SQLException e) {
+            logger.warn("Can't find bucket with id" + id);
```

Comment:
потрібно скрізь передавати помилку:
```suggestion
            logger.warn("Can't find bucket with id" + id, e);
```

Reply:
done